### PR TITLE
AUT-1253: orches modify userinfo and token endpoint

### PR DESCRIPTION
**Pull request by gn-GDS**
## What?
Instead of populating the userinfo response with data from the UserProfile table, Orchestration will use the information returned from the Authentication /userinfo endpoint. 

## Why?
We want to restrict Orchestration having access to the User tables in DynamoDB.

## Related PRs
https://github.com/alphagov/di-authentication-api/pull/3211 Orches: Create AuthCallbackHandler

### CODE REVIEW

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/services/UserInfoService.java [hunk 1]
```diff
@@ -66,28 +69,69 @@ public String calculateSubjectForAudit(AccessTokenInfo accessTokenInfo) {
...
+            }
+
+            if (accessTokenInfo.getScopes().contains(OIDCScopeValue.EMAIL.getValue())) {
+                userInfo.setEmailAddress(tmpUserInfo.getEmailAddress());
+                userInfo.setEmailVerified(tmpUserInfo.getEmailVerified());
+            }
+            if (accessTokenInfo.getScopes().contains(OIDCScopeValue.PHONE.getValue())) {
+                userInfo.setPhoneNumber(tmpUserInfo.getPhoneNumber());
+                userInfo.setPhoneNumberVerified(tmpUserInfo.getPhoneNumberVerified());
+            }
```
Comment (BeckaL) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/services/UserInfoService.java]: Should this commit and the next commit get squashed into one? They seem like logically part of the same thing. Benefits: no commenting and uncommenting in the history, unit tests appear with implementation, and easier for people to review if they review commit by commit like I do 😄

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/services/UserInfoService.java [hunk 2]
```diff
@@ -77,61 +77,9 @@ public UserInfo populateUserInfo(AccessTokenInfo accessTokenInfo) throws AccessT
...
-            }
-            if (accessTokenInfo.getScopes().contains(CustomScopeValue.GOVUK_ACCOUNT.getValue())) {
-                userInfo.setClaim("legacy_subject_id", tmpUserInfo.getClaim("legacy_subject_id"));
-            }
-            if (accessTokenInfo
-                    .getScopes()
-                    .contains(CustomScopeValue.ACCOUNT_MANAGEMENT.getValue())) {
-                userInfo.setClaim("public_subject_id", tmpUserInfo.getClaim("public_subject_id"));
-            }
+            populateInfoOrchSplitEnabled(userInfo, accessTokenInfo);
```
Comment (BeckaL) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/services/UserInfoService.java:80]: Like it, nice refactoring

**Review by BeckaL — COMMENTED** [review]
This broadly makes sense to me. Just commenting since I know others are reviewing who have more context than me, nitpicky commit comment but otherwise LGTM

**Review by DTaylorGDS — APPROVED** [review]
Looks sensible to me

**Pull request merged**